tasks/sign-binaries/sign-binaries.yaml

```diff
@@ -71,44 +71,239 @@ spec:
     - name: checksum-keytab-vol
       secret:
         secretName: $(params.checksumKeytab)
+    - name: mac-ssh-key-vol
+      secret:
+        secretName: $(params.macSSHKey)
   workspaces:
     - name: data
       description: Workspace to save the results to
   results:
+    - name: unsignedMacDigest
+      type: string
+      description: |
+        Digest used by Mac signing host to pull unsigned content via ORAS
+    - name: signedMacDigest
+      type: string
+      description: |
+        Digest used to pull signed Mac content back to pipeline via ORAS
     - name: unsignedWindowsDigest
       type: string
       description: |
-        Digest used by signing host to pull unsignged content via ORAS
+        Digest used by Windows signing host to pull unsigned content via ORAS
     - name: signedWindowsDigest
       type: string
       description: |
         Digest used to pull signed content back to pipeline via ORAS
     - name: unsignedMacDigest
       type: string
       description: |
-        Digest used by signing host to pull unsignged content via ORAS
+        Digest used by signing host to pull unsigned content via ORAS
     - name: signedMacDigest
       type: string
       description: |
         Digest used to pull signed content back to pipeline via ORAS
+    - name: binaries_path
+      type: string
+      description: |
+        Path where the final signed content is stored in the workspace
   steps:
     - name: push-unsigned-using-oras
       image: quay.io/konflux-ci/release-service-utils:e633d51cd41d73e4b3310face21bb980af7a662f
+      env:
+        - name: QUAY_USER
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: username
+        - name: QUAY_PASS
+          valueFrom:
+            secretKeyRef:
+              name: $(params.quaySecret)
+              key: password
       script: |
         #!/usr/bin/env bash
+        set -eux
+
+        CONTENT_DIR=$(workspaces.data.path)/$(params.contentDir)
+        UNSIGNED_DIR=$CONTENT_DIR/unsigned
+        MAC_CONTENT=$UNSIGNED_DIR/macos
+        WINDOWS_CONTENT=$UNSIGNED_DIR/windows
+        LINUX_CONTENT=$CONTENT_DIR/linux
+
+        mkdir -p "$MAC_CONTENT" "$WINDOWS_CONTENT" "$LINUX_CONTENT"
+        cd "$CONTENT_DIR" || exit 1
```

Comment:
The exit 1 is to make shellcheck happy, right? I think I saw that elsewhere.. @mmalina didn't we find some other way to make it pass?

Reply:
Yeah right

Reply:
Let's wait for Martin to respond tomorrow then I will either resolve or make the small adjustment and then I'll approve

Reply:
Here's the shellcheck wiki page for this: https://www.shellcheck.net/wiki/SC2164
> Rationale:
> cd can fail for a variety of reasons: misspelled paths, missing directories, missing permissions, broken symlinks and more.

The issue is that what if somewhere later you assume you're in your dir, but you never moved there. And then you do `rm *` or something.

But all of this is only a concern if you allow the script to carry on when something fails. As always, I would recommend using `set -eux` or `set -eu` in the task unless you have a good reason you don't want it. With it, the shellcheck issue disappears.